File: 프로그래머스/unrated/178871. 달리기 경주/달리기 경주.swift
import Foundation

func solution(_ players:[String], _ callings:[String]) -> [String] {
    var player = players
    var calls = callings
    var dict = [String: Int]()
    
    for i in 0..<player.count {
        dict[player[i]] = i
    }
    for i in 0..<calls.count {
        var call = calls[i]
        var idx1 = dict[call]!
        let name = player[idx1 - 1]
        player[idx1 - 1] = call
        player[idx1] = name
        dict[call]! -= 1
        dict[player[idx1]]! += 1
        
    }
    return player
}

File: 프로그래머스/unrated/178871. 달리기 경주/달리기 경주.py

def solution(players, callings):
    
    # result = players
    
    result = {string : i for i,string in enumerate(players)}
    
    for i in callings :
        index = result[i]
        if index == 0 :
            break
        else:
            result[i] -= 1
            result[players[index-1]] += 1
            players[index-1],players[index] = players[index],players[index-1]
    return players



